### DISCUSSION TITLE
SUN2000 wifi/fe dongle restart because no management connection

### DISCUSSION THREAD

**Discussion by DeWolfRobin** [wlcrs/huawei_solar#797] [Setting up the integration]
I don't have the management connection active (after a reset I wasn't able to reconnect as I'm not an installer) and now the HA integration stops working, presumably because the dongle restarts. Can I add my own ethernet to usb adaptor to the inverter? I'm not sure as to what the inverter will accept on the USB port. I could do the COM serial connection, but I would rather use TCP, as my HA system is not close to the inverter.

I get the following error after some time in HA. The setup works fine, but after a minute or two, I get this error and everything becomes unavailable

![image](https://github.com/user-attachments/assets/38111494-4963-4024-9877-ae889d9a8ab2)
This error originated from a custom integration.

Logger: custom_components.huawei_solar
Source: helpers/update_coordinator.py:360
integration: Huawei Solar (documentation, issues)
First occurred: 9:50:02 AM (2 occurrences)
Last logged: 10:06:19 AM

Timeout fetching TA23C0009151_inverter_data_update_coordinator data

**Comment by Roving-Ronin** [comment]
Huawei inverters are made to support the Huawei sDongle that has WiFi, and if you remove the cap on the bottom and look inside, they also have an ethernet connection option. If changing from WiFi to Ethernet, you need to have access to the inverter using the FusionSolar app on your phone, to configure it to use Ethernet instead (and set DHCP or Static IP also).
